File: kyu-6/_0011__Duplicate_Encoder/solution.rb
def duplicate_encode(word, new_str='')
  word = word.downcase
  word.each_char { |c| new_str += word.count(c) == 1 && '(' || ')' }
  new_str
end

p duplicate_encode('wwwdfasf');
p duplicate_encode( "(( @");
p duplicate_encode("Success"); 

File: kyu-6/_0011__Duplicate_Encoder/solution.js
function duplicateEncode(word, res = '') {
  word = word.toLowerCase();
  for (let i = 0; i < word.length; i++) {
    res += word.split(word[i]).length - 1 == 1 ? '(' : ')';
  }
  return res;
}

console.log(duplicateEncode('HII!'));
